cleanlab/filter.py

```diff
@@ -554,50 +582,90 @@ def reduce_prune_counts(prune_count_matrix, frac_noise=1.0):
     return new_mat.astype(int)
 
 
-def order_label_errors(
-        label_errors_bool,
-        psx,
-        labels,
-        sorted_index_method='normalized_margin',
-):
-    """Sorts label errors by normalized margin.
-    See https://arxiv.org/pdf/1810.05369.pdf (eqn 2.2)
-    eg. normalized_margin = prob_label - max_prob_not_label
+def find_argmax_not_equal_given_label(labels, psx, multi_label=False):
+    """This is the simplest baseline approach. Just consider
+    anywhere argmax != labels as a label error.
 
     Parameters
     ----------
-    label_errors_bool : np.array (bool)
-      Contains True if the index of labels is an error, o.w. false
+    labels : np.array
+        A discrete vector of noisy labels, i.e. some labels may be erroneous.
 
     psx : np.array (shape (N, K))
-      P(s=k|x) is a matrix with K probabilities for all N examples x.
-      This is the probability distribution over all K classes, for each
-      example, regarding whether the example has label s==k P(s=k|x). psx
-      should computed using 3 (or higher) fold cross-validation.
+        P(label=k|x) is a matrix with K (noisy) probabilities for each of the
+        N examples x. This is the probability distribution over all K classes,
+        for each example, regarding whether the example has label labels==k P(labels=k|x).
+        psx should have been computed using 3 (or higher) fold cross-validation.
+
+    multi_label : bool
+        Set to True if labels is multi-label (list of lists, or np.array of np.array)
+
+    Returns
+    -------
+        A boolean mask that is true if the example belong
+        to that index is label error."""
+
+    assert (len(psx) == len(labels))
+    if multi_label:
+        # Todo: This needs to be tested.
+        return np.array(
+            [all(np.argsort(psx[i])[:len(j)] == sorted(j)) for i, j in enumerate(labels)])
+    return np.argmax(psx, axis=1) != np.asarray(labels)
+
+
+def find_label_issues_using_argmax_confusion_matrix(
+    labels,
+    psx,
+    calibrate=True,
+    filter_by='prune_by_noise_rate',
+):
+    """This is a baseline approach. That uses the confusion matrix
+    of argmax(psx) and labels as the confident joint and then uses cleanlab
+    (confident learning) to find the label issues using this matrix.
+
+    The only difference between this and find_label_issues is that it uses the confusion matrix
+    based on the argmax and given label instead of using the confident joint from
+    `count.compute_confident_joint`.
+
+    This method does not support multi-label labels.
+
+    Parameters
+    ----------
 
     labels : np.array
-      A binary vector of labels, which may contain label errors.
+        A discrete vector of noisy labels, i.e. some labels may be erroneous.
```

Comment:
clarify: for dataset with `K` classes, labels must be in `{0,1,...,K-1}`?

Reply:
clarified everywhere